docs/guides/contexts_expressions.md

```diff
@@ -0,0 +1,390 @@
+**operations** that can be reduced to a single value. For example, `1 > 5` is an
+expression that always reduces to the value `False` when evaluated.  `x > 5 and
+y == 'hello'` is also an expression that reduces to a boolean value; however, in
+order to evaluate that expression, we need to be able to fill in the values for
+the variable **names** "`x`" and "`y`". Those values are provided by some
+**context** (or "namespace"), which maps the variable names to their values.
+
+The value of an `expression` depends on the context in which it is evaluated.
+
+```python
+In [1]: expression = "x > 5 and y == 'hello'"
+
+In [2]: context_a = {'x': 7, 'y': 'hello'}
+
+In [3]: context_b = {'x': 8, 'y': 'howdie!'}
+
+In [4]: eval(expression, context_a)
+Out[4]: True
+
+In [5]: eval(expression, context_b)
+Out[5]: False
+```
+
+## Napari expressions
+
+In napari, we'd like to be able to capture the concept of some condition being
+`True` or `False`, *prior* to actually having the context required to evaluate
+it.  For example, a plugin (or napari itself) might want to stipulate that a
+given function should only be enabled when "the active layer has at least 3
+dimensions".
+
+At runtime, in Python code, this might be captured by the expression:
+
+```python
+viewer.layers.selection.active.data.ndim >= 3
+```
+
+... however, if you don't have access to the future `viewer` instance, that
+doesn't work.  So napari has the concept of `Expr` objects that represent an
+expression "without a context", to be evaluated later.
+
+```{tip}
+napari's `Expr` class subclasses from
+[`ast.AST`](https://docs.python.org/3/library/ast.html#ast.AST) and shares many
+similarities with the `body` of
+[`ast.Expr`](https://docs.python.org/3/library/ast.html#ast.Expr). However, for
+the sake of evaluation safety, napari's `Expr` only supports a subset of
+operations, omitting things like function calls, generators, comprehensions, and
+collections. It's not important to fully understand ASTs to use napari
+expressions, but for a good introduction to Python's abstract syntax tree (AST)
+module, see https://greentreesnakes.readthedocs.io.
+```
+
+A string expression can be converted to a napari expression with
+`parse_expression`:
+
+```python
+In [6]: from napari.utils.context import parse_expression
+
+In [7]: expr = parse_expression("x > 5 and y == 'hello'")
+
+In [8]: expr
+Out[8]:
+BoolOp(
+  op=And(),
+  values=[
+    Compare(
+      left=Name(id='x', ctx=Load()),
+      ops=[Gt()],
+      comparators=[Constant(value=5)]
+    ),
+    Compare(
+      left=Name(id='y', ctx=Load()),
+      ops=[Eq()],
+      comparators=[Constant(value='hello')]
+    )
+  ]
+)
+```
+
+The expression object can be evaluated using `eval`
+
+```python        
+In [9]: expr.eval({'x': 7, 'y': 'hello'})
+Out[9]: True
+```
+
+and it can also be combined with other expressions & constants using
+**operators**:
+
+```python
+In [10]: new_expr = expr & (7 > 10)  # always False
```

Comment:
even if it's neat that you can do that with Python constant, I think it might be better for a tutorial to 
do 
```
In [10]: other_expr = parse_expr("7 > 10")

In [11]: new_expr = expr & other_expr
```

That does work in all the case, even when what is on the RHS of `&` is not purely operator and constants. 
Yes it's more typing and strictly unnecessary but that does not mix concepts. 

Just a though depending on the level of your audiance.

Reply:
Reading it more, I think that's ok to leave the `new_expr = expr & (7 > 10)` as this is for people to understand inner napari working.